### PR TITLE
Improve Eclipse-compatible restart file. Output correct Guiderate control and FIELD level control

### PR DESCRIPTION
**Pull request by jalvestad**
This pull request contains improvements of the Eclipse-compatible restart file. The improvements are:
1. Output of Guiderate control code for general guiderate input.

2. Output of correct values for IGRP (WInjActiveCMode, GInjActiveCMode, GConInjeWInjCMode and GConInjeGInjCMode)

The pull request is ready for review and merge after improvements based on comments from reviewers.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/output/eclipse/AggregateGroupData.cpp [hunk 1]
```diff
@@ -447,36 +459,38 @@ void productionGroup(const Opm::Schedule&     sched,
...
+        // group is available for higher level control, but is currently constrained by own limits
+        iGrp[nwgmax + IGroup::ProdHighLevCtrl] = -1;
+        if ((deck_cmode != Opm::Group::ProductionCMode::FLD) && !group.productionGroupControlAvailable()) {
+            //group is not free to respond to higher level control)
+            iGrp[nwgmax + IGroup::ProdHighLevCtrl] = 0;
+        } else if (cgroup && ((active_cmode == Opm::Group::ProductionCMode::FLD) || (active_cmode == Opm::Group::ProductionCMode::NONE))) {
+            //a higher level group control is active constraint
+            if ((deck_cmode != Opm::Group::ProductionCMode::FLD) && (deck_cmode != Opm::Group::ProductionCMode::NONE)) {
+                iGrp[nwgmax + IGroup::ProdHighLevCtrl] = cgroup->insert_index();
+            } else if ((deck_cmode == Opm::Group::ProductionCMode::FLD) && (prod_guide_rate_def != Opm::Group::GuideRateProdTarget::NO_GUIDE_RATE)) {
```
Comment (joakim-hove) [inline src/opm/output/eclipse/AggregateGroupData.cpp:474]: This is heacy reading - is it possible to summarize in a few sentences of english?

Comment (jalvestad) [reply src/opm/output/eclipse/AggregateGroupData.cpp:474]: The main changes are to handle the FIELD group level control correctly. That consists of two main changes:

1. The controlling group (production / injection) can be the FIELD level as well as a higher group level (controlGroup and injectionControlGroup functions)

2. A group being controlled by a group at a higher level can also be controlled from the "FIELD" level.

**Comment by jalvestad** [comment]
I have checked the test failures and they are due to the intended changes to the IGRP - array and are correct as far as I can judge.

**Comment by joakim-hove** [comment]
jenkins build this update_data please

**Comment by joakim-hove** [comment]
jenkins build this opm-tests=617 please

**Comment by bska** [comment]
jenkins build this opm-tests=617 please

**Comment by bska** [comment]
Merging this now.  Thanks a lot!

**Pull request merged**